usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -109,15 +109,119 @@ def get_total_results(keyword, sub_index, retries=3):
         return None
 
 
+def category_query_for_all(keyword, sub_index):
+    query = {
+              "query": {
+                "bool": {
+                  "must": [
+                      {"query_string": {"query": keyword}}],
+                  "filter": {
+                    "terms": {
+                      "award_category": sub_index
+                    }
+                  }
+                }
+              }
+            }
+    return query
+
+
+def category_query_for_contract_nulls(keyword, sub_index):
+    query = {"query": {
+                    "bool": {
+                        "must": [
+                            {"query_string": {"query": keyword}},
+                            {"match": {"pulled_from": "IDV"}}
+                        ]
+                        }
+                    }
+             }
+    return query
+
+
+def clean_sub_index(sub_index):
+    lookup = {"loans": ["loans"], "other": ["other"], "contracts":
+              ["contract"], "direct_payments": ["direct", "payment"], "grants": ["grant"], "insurance": ["insurance"]}
+    try:
+        return lookup[(sub_index)]
+    except KeyError:
+        logger.error('Unexpected Response')
+        return None
+
+
+def get_total_results_by_award_category(keyword, sub_index, retries=3):
```

Comment:
Can you look at this function and perhaps think of ways to reduce the slightly-different code so that this function can be simplified?

Reply:
Hey Tony, I completely refactored the code and optimized it. So those several redundant checks on the requests are no longer an issue, as there is only one request.